src/MongoDB.Driver/PipelineDefinitionBuilder.cs

```diff
@@ -1084,6 +1084,27 @@ public static class PipelineDefinitionBuilder
             return pipeline.AppendStage(PipelineStageDefinitionBuilder.SortByCount(value, translationOptions));
         }
 
+        /// <summary>
+        /// Appends a $unionWith stage to the pipeline.
+        /// </summary>
+        /// <typeparam name="TInput">The type of the input documents.</typeparam>
+        /// <typeparam name="TOutput">The type of the output documents.</typeparam>
+        /// <typeparam name="TWith">The type of the with collection documents.</typeparam>
+        /// <param name="pipeline">The pipeline.</param>
+        /// <param name="withCollection">The with collection.</param>
+        /// <param name="withPipeline">The with pipeline.</param>
+        /// <returns>The stage.</returns>
+        public static PipelineDefinition<TInput, TOutput> UnionWith<TInput, TOutput, TWith>(
```

Comment:
Sorry for suggesting the wrong order for the type parameters. I now realize that in this class we always put TOuput last, so this should be:

    public static PipelineDefinition<TInput, TOutput> UnionWith<TInput, TWith, TOutput>